pkg/dp/apisix/upstream.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package apisix
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/apache/apisix-control-plane/pkg/utils"
+	"github.com/gxthrj/seven/conf"
+)
+
+func (upstream *Upstream) Add() (*UpstreamResponse, error) {
+	baseUrl := conf.BaseUrl
+	url := fmt.Sprintf("%s/upstreams", baseUrl)
+	if b, err := json.Marshal(upstream); err != nil {
```

Comment:
ditto

Reply:
This is some style of golang, many gophers like this style